app/containers/Organisations/Facility.tsx

```diff
@@ -44,7 +44,11 @@ export const FacilityComponent: React.FunctionComponent<Props> = ({
   const applyButton = () => {
     if (!applicationId) {
       return (
-        <Button variant="primary" onClick={startApplication}>
+        <Button
+          variant="primary"
+          data-action="apply"
```

Comment:
I'm not convinced about adding a property for testing purposes. @wenzowski  any opinion on that?

Reply:
We can make it `data-test-` or `test-data-` and remove these attr when building prod version

Reply:
If @wenzowski approves, then I would go for `test-data-action`. That should also be documented in our developer guide. I started a section on tests in #205

Reply:
Let's please not add production-path code to support testing.

Why is the `data-action` necessary?  The button had an `onClick`...why can't we bind an event handler?

Reply:
@wenzowski this is attribute for testing, not for action handling.

Reply:
@dimak1 what are you testing then, if not an action?

Reply:
Testing dynamic button. If application is completed it should show `View` button, otherwise, `Resume`. Updated tests.